src/modm/platform/spi/stm32_uart/uart_spi_master.hpp.in

```diff
@@ -72,28 +72,9 @@ public:
 	static void
 	initialize()
 	{
-%% if target["family"] != "f1"
-		constexpr UartBase::OversamplingMode oversample =
-				UartBaudrate::getOversamplingMode(SystemClock::Usart{{ id }}, baudrate);
-%% endif
-		UsartHal{{ id }}::initializeWithBrr(
-				UartBaudrate::getBrr<SystemClock::Usart{{ id }}, baudrate, tolerance>(),
-%% if target["family"] == "f1"
-				UsartHal{{ id }}::Parity::Disabled);
-%% else
-				UsartHal{{ id }}::Parity::Disabled,
-				oversample);
-%% endif
-
-%% if extended
-		UsartHal{{ id }}::disableOperation();
-%% endif
+		UsartHal{{ id }}::initialize<SystemClock, baudrate, tolerance>(UsartHal{{ id }}::Parity::Disabled, UsartHal{{ id }}::WordLength::Bit8);
 		UsartHal{{ id }}::setLastBitClockPulse(UsartHal{{ id }}::LastBitClockPulse::Output);
 		UsartHal{{ id }}::setSpiClock(UsartHal{{ id }}::SpiClock::Enabled);
-%% if extended
-		UsartHal{{ id }}::enableOperation();
-%% endif
```

Comment:
Why was it previously necessary but is no longer necessary to stop operation before modifying the clock settings?

Reply:
It wasn't, all of the three operations each do enable and disable.

Reply:
Yes, altough our API is weird too, it should be able to configure the entire peripheral in call, since modifying any of these settings during UART operation is not going to make sense without stopping, particularly with buffers involved. Another PR though.